src/librustc_resolve/resolve_imports.rs

```diff
@@ -689,13 +691,21 @@ impl<'a, 'b:'a, 'c: 'b> ImportResolver<'a, 'b, 'c> {
                 // If the error is a single failed import then create a "fake" import
                 // resolution for it so that later resolve stages won't complain.
                 self.import_dummy_binding(import);
+                if prev_root_id.as_u32() != 0 &&
+                    prev_root_id.as_u32() != import.root_id.as_u32() &&
+                    !error_vec.is_empty(){
+                    // in case of new import line, throw diagnostic message
+                    // for previous line.
+                    self.throw_unresolved_import_error(error_vec.clone(), None);
+                    error_vec.clear();
```

Comment:
Here I feel using `std::mem::swap` would be a better idea:

```
let mut empty_vec = vec![];
swap(empty_vec, error_vec);
self.throw_unresolved_import_error(empty_vec, None);
```

That way you're not cloning `error_vec` just to discard its contents immediately after.

Reply:
done!